testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -6178,6 +6178,34 @@ private static void checkIf(SqlOperatorFixture f) {
     f.checkNull("str_to_map('a:1,b:2,c:3', ',',null)");
   }
 
+  /** Tests {@code SUBSTRING_INDEX} function from Spark. */
+  @Test void testSubstringIndexFunc() {
```

Comment:
This function performs a case-sensitive match when searching for delim. I think we should add a test to make a comparison.

Reply:
OK, I will add two cases to make a comparsion.
```
f.checkString("substring_index('aBc', 'B', -1)",
        "c", "VARCHAR(3) NOT NULL");
    f.checkString("substring_index('aBc', 'b', -1)",
        "aBc", "VARCHAR(3) NOT NULL");
```